### PR TITLE
#1741 Split the read functionality into a stand-alone module. 

### PR DESCRIPTION
**Pull request by hiker**
Removed…`netcdf` from the names in the reading module and driver so the names are more appropriate for non-netcdf drivers.

### CODE REVIEW

**Comment by hiker** [comment]
This is a quite small PR that moves the read functionality out of the psydata libraries into a stand-alone module. This will make it easier to compile a driver without infrastructure (since the psydata libraries add dependencies to the infrastructure libraries).

File: examples/gocean/eg5/extract/Makefile [hunk 1]
```diff
@@ -95,20 +96,18 @@ $(NAME): $(INF_LIB) $(EXTRACT_DIR)/lib_kernel_data_netcdf.a $(KERNELS) alg.o psy
 	$(INF_LIB)                                \
 	$$(nf-config --flibs)
 
-$(DRIVER_INIT):	$(KERNELS) $(DRIVER_INIT).o \
-				$(EXTRACT_DIR)/lib_kernel_data_netcdf.a
+$(DRIVER_INIT):	$(KERNELS) $(DRIVER_INIT).o
```
Comment (arporter) [inline examples/gocean/eg5/extract/Makefile:99]: I think this is missing a dependence on `$(READ_DIR)/read_kernel_data_mod.o`. I get (when doing `cd examples; make allclean; make compile`):

    gfortran -g -O0 $(nf-config --fflags) -I../../../../external/dl_esm_inf/finite_difference/src -I../../../../lib/extract/netcdf/dl_esm_inf -I../../../../lib/extract/netcdf -c driver-main-init.f90
    driver-main-init.f90:2:6:

    2 |   use read_kernel_data_mod, only : ReadKernelDataType
      |      1
    Fatal Error: Cannot open module file ‘read_kernel_data_mod.mod’ for reading at (

File: lib/extract/netcdf/.gitignore [hunk 2]
```diff
@@ -1,2 +1,2 @@
 extract_netcdf_base.f90
-
+read_kernel_data_mod.f90
```
Comment (arporter) [inline lib/extract/netcdf/.gitignore:2]: Please add a newline to the end of this file.

File: examples/gocean/eg5/extract/Makefile [hunk 3]
```diff
@@ -95,20 +96,18 @@ $(NAME): $(INF_LIB) $(EXTRACT_DIR)/lib_kernel_data_netcdf.a $(KERNELS) alg.o psy
...
 	$(F90) $(KERNELS) $(DRIVER_INIT).o \
 	-o $(DRIVER_INIT)            \
-	-L$(EXTRACT_DIR) -l_kernel_data_netcdf \
+	$(READ_DIR)/read_kernel_data_mod.o \
 	$(INF_LIB)     \
 	$$(nf-config --flibs)
 
-$(DRIVER_UPDATE): $(KERNELS) $(DRIVER_UPDATE).o \
-				$(EXTRACT_DIR)/lib_kernel_data_netcdf.a
+$(DRIVER_UPDATE): $(KERNELS) $(DRIVER_UPDATE).o
```
[the remark this answers is not in the document]
Comment (hiker) [reply examples/gocean/eg5/extract/Makefile]: Done.

File: lib/extract/netcdf/.gitignore (hunk 2 above)
Comment (hiker) [reply lib/extract/netcdf/.gitignore]: Done.

**Pull request merged**